src/naksu/boxversion/boxversion.go

```diff
@@ -0,0 +1,202 @@
+  boxTimestamp int64
+  updateStarted int64
+}
+
+// Global cache for GetVagrantBoxAvailVersionDetails()
+var vagrantBoxAvailVersionDetailsCache lastBoxAvail
+var vagrantBoxAvailVersionDetailsCacheTimeout int64 = 5*60	// in seconds (5 minutes)
+
+
+// GetVagrantFileVersion returns a human-readable localised version string
+// for a given Vagrantfile (with "" defaults to ~/ktp/Vagrantfile)
+func GetVagrantFileVersion (vagrantFilePath string) string {
+	if vagrantFilePath == "" {
+			vagrantFilePath = mebroutines.GetVagrantDirectory() + string(os.PathSeparator) + "Vagrantfile"
+	}
+
+	boxString, boxVersion, err := GetVagrantFileVersionDetails(vagrantFilePath)
+	if err != nil {
+		mebroutines.LogDebug(fmt.Sprintf("Could not read from %s", vagrantFilePath))
+		return ""
+	}
+
+	boxType := GetVagrantBoxType(boxString)
+
+  versionString := fmt.Sprintf("%s (v%s)", boxType, boxVersion)
+  mebroutines.LogDebug(fmt.Sprintf("GetVagrantFileVersion returns: %s", versionString))
+
+	return versionString
+}
+
+// GetVagrantFileVersionDetails returns version string (e.g. "digabi/ktp-qa") and
+// version number (e.g. "66") from the given vagrantFilePath
+func GetVagrantFileVersionDetails (vagrantFilePath string) (string, string, error) {
+	fileContent, err := ioutil.ReadFile(filepath.Clean(vagrantFilePath))
+	if err != nil {
+		mebroutines.LogDebug(fmt.Sprintf("Could not read from %s", vagrantFilePath))
+		return "", "", err
+	}
+
+	boxRegexp := regexp.MustCompile(`config.vm.box = "(.+)"`)
+	versionRegexp := regexp.MustCompile(`vb.name = ".+v(\d+)"`)
+
+	boxMatches := boxRegexp.FindStringSubmatch(string(fileContent))
+	versionMatches := versionRegexp.FindStringSubmatch(string(fileContent))
+
+	if len(boxMatches) == 2 && len(versionMatches) == 2 {
+    mebroutines.LogDebug(fmt.Sprintf("GetVagrantFileVersionDetails returns: [%s] [%s]", boxMatches[1], versionMatches[1]))
+		return boxMatches[1], versionMatches[1], nil
+	}
+
+	return "", "", errors.New("did not find values from vagrantfile")
+}
+
+
+// GetVagrantBoxAvailVersion returns a human-readable localised version string
+// for a vagrant box available with update
+func GetVagrantBoxAvailVersion () string {
+	boxString, boxVersion, err := GetVagrantBoxAvailVersionDetails()
+	if err != nil {
+		mebroutines.LogDebug("Could not get available version string")
+		return ""
+	}
+
+	boxType := GetVagrantBoxType(boxString)
+
+  versionString := fmt.Sprintf("%s (v%s)", boxType, boxVersion)
+  mebroutines.LogDebug(fmt.Sprintf("GetVagrantBoxAvailVersion returns: %s", versionString))
+
+	return versionString
+}
+
+// GetVagrantBoxAvailVersionDetails gets info about available vagramt box
+// from ReallyGetVagrantBoxAvailVersionDetails() or global vagrantBoxAvailVersionDetailsCache
+func GetVagrantBoxAvailVersionDetails () (string, string, error) {
+  boxString := ""
+  boxVersion := ""
+  var boxError error
+
+  // There is a avail version fetch going on
+  for vagrantBoxAvailVersionDetailsCache.updateStarted != 0 {
```

Comment:
This for loop has a risk of being eternal, if the condition is never met for some reason.

Reply:
Would exiting the loop/throwing an error after a number of tries be more safe? Here's an example of limiting iterations in a for loop: https://tour.golang.org/flowcontrol/1

Reply:
I agree, having something like 240 iteration limit here would be useful (2 minutes)

Reply:
:+1: